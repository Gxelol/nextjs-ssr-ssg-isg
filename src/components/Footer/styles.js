import styled, { css } from 'styled-components';

import { Container as Text } from '../Text/styles';
import { Container as Section } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.primaryColor};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${Text} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${Section} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
`;
