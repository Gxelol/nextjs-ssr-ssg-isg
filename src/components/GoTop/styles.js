import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    z-index: 6;
    opacity: 0.8;
  `}
`;
