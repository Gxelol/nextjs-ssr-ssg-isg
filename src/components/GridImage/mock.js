export default {
  background: false,
  title: 'Gallery',
  description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Eum asperiores saepe impedit illo recusandae! Et assumenda vitae quos.`,
  grid: [
    {
      altText: 'Test 1',
      srcImg: 'https://source.unsplash.com/random/360x360?r=1',
    },
    {
      altText: 'Test 2',
      srcImg: 'https://source.unsplash.com/random/360x360?r=2',
    },
    {
      altText: 'Test 3',
      srcImg: 'https://source.unsplash.com/random/360x360?r=3',
    },
    {
      altText: 'Test 4',
      srcImg: 'https://source.unsplash.com/random/360x360?r=4',
    },
    {
      altText: 'Test 5',
      srcImg: 'https://source.unsplash.com/random/360x360?r=5',
    },
    {
      altText: 'Test 6',
      srcImg: 'https://source.unsplash.com/random/360x360?r=6',
    },
  ],
};
