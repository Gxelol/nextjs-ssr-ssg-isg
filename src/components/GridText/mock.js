export default {
  background: false,
  title: 'My Grid',
  description: `Lorem ipsum, dolor sit amet consectetur adipisicing elit. Eum asperiores saepe impedit illo recusandae! Et assumenda vitae quos, deserunt odit at pariatur totam ad culpa voluptatem ex placeat quod porro.`,
  grid: [
    {
      title: 'Test 1',
      description:
        'Morbi hendrerit nec augue ac placerat. Duis vestibulum augue turpis. Nulla consectetur justo ac justo aliquet, vel auctor augue condimentum.',
    },
    {
      title: 'Test 2',
      description:
        'Morbi hendrerit nec augue ac placerat. Duis vestibulum augue turpis. Nulla consectetur justo ac justo aliquet, vel auctor augue condimentum.',
    },
    {
      title: 'Test 3',
      description:
        'Morbi hendrerit nec augue ac placerat. Duis vestibulum augue turpis. Nulla consectetur justo ac justo aliquet, vel auctor augue condimentum.',
    },
  ],
};
