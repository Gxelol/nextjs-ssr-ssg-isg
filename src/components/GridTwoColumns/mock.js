export default {
  title: 'Grid Two Columns',
  text: `Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod voluptatibus dolores obcaecati reprehenderit officiis libero recusandae fuga amet molestiae commodi, provident alias numquam doloribus ducimus debitis eius nemo laboriosam? Sapiente!`,
  srcImg: 'assets/images/javascript.svg',
};
