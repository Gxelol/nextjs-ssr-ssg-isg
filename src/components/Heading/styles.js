import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.small};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
    `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.big};
    `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.huge};
    ${mediaFont(theme)}
    `,
};

const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.fonts.sizes.big};
  }
`;

const titleCase = (uppercase) => css`
text-transform: ${uppercase ? 'uppercase' : 'none'}
`;

export const Title = styled.h1`
  ${({ theme, darkColor, size, uppercase }) => css`
    color: ${darkColor ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`;
