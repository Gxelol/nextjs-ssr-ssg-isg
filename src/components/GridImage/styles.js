import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
    gap: ${theme.spacings.large};

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;
    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
