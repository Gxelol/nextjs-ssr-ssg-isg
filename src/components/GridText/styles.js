import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';
import { Container as Text } from '../Text/styles';

export const Container = styled.section`
  ${({ theme }) => css`
    ${Text} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const GridContainer = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(28rem, 1fr));
    gap: ${theme.spacings.large};
    overflow: hidden;
    width: 100%;

    @media ${theme.media.lteSmall} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
  ${Heading} {
    position: relative;
    left: 5rem
  }

  ${Heading}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
  }
  `}
`;
