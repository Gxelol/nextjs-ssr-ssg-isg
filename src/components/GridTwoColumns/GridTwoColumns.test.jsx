import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { GridTwoColumns } from '.';
import mock from './mock';

describe('<GridTwoColumns />', () => {
  it('should render grid two columns', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
