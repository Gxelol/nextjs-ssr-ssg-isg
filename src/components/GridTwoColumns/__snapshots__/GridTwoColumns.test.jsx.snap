// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridTwoColumns /> should render grid two columns 1`] = `
.c5 {
  color: #0A1128;
  font-size: 5.8rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4rem;
}

.c7 {
  width: 100%;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

.c0 {
  background: #ffffff;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:76.8rem) and (min-width:38.5rem) {
  .c5 {
    font-size: 4.6rem;
  }
}

@media (max-width:38.4rem) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:38.4rem) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <section
    class="c0"
    id=""
  >
    <section
      class="c1"
    >
      <section
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid Two Columns
          </h2>
          <div
            class="c6"
          >
            Lorem, ipsum dolor sit amet consectetur adipisicing elit. Quod voluptatibus dolores obcaecati reprehenderit officiis libero recusandae fuga amet molestiae commodi, provident alias numquam doloribus ducimus debitis eius nemo laboriosam? Sapiente!
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid Two Columns"
            class="c7"
            src="assets/images/javascript.svg"
          />
        </div>
      </section>
    </section>
  </section>
</div>
`;
